agent-control/src/cli/on_host/config_gen/identity.rs

```diff
@@ -0,0 +1,338 @@
+//! This module provides the functions to handle identity creation when generating Agent Control configuration
+
+use std::{path::PathBuf, time::Duration};
+
+use nr_auth::{
+    TokenRetriever,
+    authenticator::HttpAuthenticator,
+    http::{client::HttpClient, config::HttpConfig},
+    key::{
+        creator::KeyType,
+        local::{KeyPairGeneratorLocalConfig, LocalCreator},
+    },
+    system_identity::{
+        generator::L2SystemIdentityGenerator,
+        iam_client::http::HttpIAMClient,
+        input_data::{
+            SystemIdentityCreationMetadata, SystemIdentityInput, environment::NewRelicEnvironment,
+            output_platform::OutputPlatform,
+        },
+    },
+    token::{Token, TokenType},
+    token_retriever::TokenRetrieverWithCache,
+};
+use tracing::info;
+
+use crate::{cli::error::CliError, http::config::ProxyConfig};
+
+use super::Args;
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(3);
+const DEFAULT_RETRIES: u8 = 3;
+
+/// Represents a key-based identity to be used in Agent Control configuration.
+pub struct Identity {
+    pub client_id: String,
+    pub private_key_path: PathBuf,
+}
+
+/// Provides a key-based identity considering the supplied args.
+pub fn provide_identity(args: &Args) -> Result<Identity, CliError> {
+    let environment = NewRelicEnvironment::from(args.region);
+    build_identity(args, environment)
+}
+
+/// Helper to allow injecting testing urls when building the identity.
+fn build_identity(args: &Args, environment: NewRelicEnvironment) -> Result<Identity, CliError> {
+    // Already provided identity
+    if !args.auth_client_id.is_empty() {
+        return Ok(Identity {
```

Comment:
Maybe a log would be nice. It's hard for me to know what it's too much and what it's too little. I see you added `info!("Generating System Identity");`. Maybe a log in this other path would be nice.

```
info!(Using passed identity);
```

Reply:
I wanted to keep logs simple but I probably went too far! I'm adding some details about how the identity is generated.

Reply:
I added more logs here: [1f4d4da](https://github.com/newrelic/newrelic-agent-control/pull/1779/commits/1f4d4dac5e9342324e519238fb64e6b9dcaad1a8)